gh_32-bit-build-smoke is failing when compiled with the cflags -Wall -Werror

https://build.gluster.org/job/gh_32-bit-build-smoke/3812/

Error:
```
14:53:30 make[3]: Entering directory '/src/libglusterfs/src'
14:53:30   CC       libglusterfs_la-dict.lo
14:53:30   CC       libglusterfs_la-xlator.lo
14:53:30 dict.c: In function 'dict_serialize_value_with_delim_lk':
14:53:30 dict.c:2995:13: error: the comparison will always evaluate as 'true' for the address of 'key' will never be NULL [-Werror=address]
14:53:30  2995 |         if (!pair->key || !pair->value) {
14:53:30       |             ^
14:53:30 In file included from dict.c:19:
14:53:30 glusterfs/dict.h:100:10: note: 'key' declared here
14:53:30   100 |     char key[];
14:53:30       |          ^~~
14:53:31   CC       libglusterfs_la-logging.lo
14:53:31 cc1: all warnings being treated as errors
14:53:31 make[3]: *** [Makefile:1080: libglusterfs_la-dict.lo] Error 1
```

--- a/libglusterfs/src/dict.c
+++ b/libglusterfs/src/dict.c
@@ -2992,7 +2992,7 @@ dict_serialize_value_with_delim_lk(dict_t *this, char *buf, int32_t *serz_len,
             goto out;
         }
 
-        if (!pair->key || !pair->value) {
+        if (!pair->value) {
             gf_smsg("dict", GF_LOG_ERROR, 0, LG_MSG_KEY_OR_VALUE_NULL, NULL);
             goto out;
         }

